streamlit_fesion/frontend/src/MyComponent.tsx

```diff
@@ -21,108 +16,39 @@ const MyComponent: React.VFC = () => {
+      const workerProxy = new WorkerProxy({
+        funcName: imageFilterPyFuncName,
+        funcDefPyCode: imageFilterPyFuncDefCode,
+        requirements: filterDepPackages,
+      });
 
-    if (pyodide == null) {
+      setWorkerProxy(workerProxy);
+    },
+    [
+      // No deps because workerProxy should not be re-created.
+    ]
+  );
+  useEffect(() => {
+    if (workerProxy == null) {
       return;
     }
 
-    const filterDepPackages: string[] | null = JSON.parse(
-      imageFilterDepPackagesJson
-    );
-
-    (async () => {
-      // Import NumPy, which is used in the wrapper script.
-      await pyodide
-        .loadPackage(["numpy"])
-        .then(() =>
-          pyodide.runPythonAsync(`import numpy as ${NUMPY_GLOBAL_ALIAS}`)
-        );
-
-      // Load packages used in the user-defined filter function.
-      await pyodide.loadPackagesFromImports(imageFilterPyFuncDefCode);
-      if (filterDepPackages) {
-        await pyodide.loadPackage(filterDepPackages);
-      }
-
-      // TODO: Delete the previous filter func by running "del {func_name}" to avoid memory leak.
-
-      // Run the Python code including the user-defined filter function.
-      await pyodide.runPythonAsync(imageFilterPyFuncDefCode);
-
-      const filterFn: ImageDataFilter = async (imageData) => {
-        // TODO: Run in WebWorker
-        /* eslint-disable @typescript-eslint/ban-ts-comment */
-        // @ts-ignore
-        self.fesionImageWidth = imageData.width;
-        // @ts-ignore
-        self.fesionImageHeight = imageData.height;
-        // @ts-ignore
-        self.fesionImageData = imageData.data;
-        /* eslint-enable */
+    const filterDepPackages: string[] =
+      JSON.parse(imageFilterDepPackagesJson) || [];
 
-        try {
-          await pyodide.runPythonAsync(`
-          from js import fesionImageWidth, fesionImageHeight, fesionImageData  # Import from JS globals
-
-          input_image4chan = ${NUMPY_GLOBAL_ALIAS}.asarray(fesionImageData.to_py()).reshape((fesionImageHeight, fesionImageWidth, 4)) # 4 channels (RGBA)
-          input_image = input_image4chan[:,:,:3]
-
-          output_image = ${imageFilterPyFuncName}(input_image)
-
-          if ${NUMPY_GLOBAL_ALIAS}.issubdtype(output_image.dtype, ${NUMPY_GLOBAL_ALIAS}.floating):
-              output_image = (output_image * 255).astype(${NUMPY_GLOBAL_ALIAS}.uint8)
-
-          output_alpha = ${NUMPY_GLOBAL_ALIAS}.full((fesionImageHeight, fesionImageWidth, 1), fill_value=255, dtype=${NUMPY_GLOBAL_ALIAS}.uint8)
-          output_image4chan = ${NUMPY_GLOBAL_ALIAS}.concatenate((output_image, output_alpha), axis=2).copy()
-
-          output_height, output_width = output_image4chan.shape[:2]
-          `);
-        } catch (err: any) {
-          if (err.name === "PythonError") {
-            const serializableError = {
-              stack: err.stack,
-              message: err.message,
-            };
-            setComponentValue({ pythonError: serializableError });
```

Comment:
TODO: Use `pythonError` return value to delegate the Python-level errors to the Streamlit app script.